### PR TITLE
[Dashboard] Add a better error message when the dashboard fils to start

### PR DESCRIPTION
**Pull request by rkooo567**
Signed-off-by: SangBin Cho <rkooo567@gmail.com>

## Why are these changes needed?

Currently, when the dashboard fails to start even before it initializes the logger, the error message is disappeared because all the stdout/stderr (which contains the exception) is redirected to DEVNULL. This PR fixes the issue by redirecting stdout + stderr to the new dashboard.err.  (we are doing the same thing log monitor).

The dashboard error printing logic is as follow.
1. try starting a dashboard.
2. If it doesn't start, print .log
3. if it cannot find .log file (which means the dashboard crashes even before logger is initialized), it prints .err
4. If it also fails to print .err, it falls back to the current error. This should theoretically never happen 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
Note I tried adding a test, but it was pretty challenging... (we should add manual failure using the env var to the code) I manually verified it works.

File: doc/source/ray-observability/ray-logging.rst [hunk 1]
```diff
@@ -152,12 +152,12 @@ A new Ray instance creates a new session ID to the temp directory. The latest se
 
 Here's a Ray log directory structure. Note that ``.out`` is logs from stdout/stderr and ``.err`` is logs from stderr. The backward compatibility of log directories is not maintained.
 
-- ``dashboard.log``: A log file of a Ray dashboard.
+- ``dashboard.[log|out|err]``: A log file of a Ray dashboard. ``log.`` file contains logs generated from the dashboard's logger. ``.out`` and ``.err`` are stdout and stderr printed from the dashboard. They are usually empty except when the dashboard crashes unexpectedly.
```
Comment (rickyyx) [inline doc/source/ray-observability/ray-logging.rst:155]: Will there be `out` files for dashboard and log_monitor? Thought we are redirecting stdout and stderr to a single `.err` file.

**Review by rickyyx — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/gifs/share-very-discover-10uEX5kfeodYgo">GIPHY</a></div>

File: doc/source/ray-observability/ray-logging.rst (hunk 1 above)
Comment (rkooo567) [reply doc/source/ray-observability/ray-logging.rst:155]: ah good catch. Fixign it now..

**Review by maxpumperla — APPROVED** [review]
LGTM, thanks

**Pull request merged**